lukemckinstry/ebus-breakable-toy
# Feature/lcm/route batch api endpoint

Creates API endpoint to support managing routes via a batch process running outside the application.

### Endpoints:
`agency/<str:agency_pk>/batch/route` --> list or create a route
`route/batch/<str:pk>/` --> update a route

### Testing Instructions:
 - Obtain an API access token with the following payload filled in with credentials for a user you have created in your dev environment: 
```json
curl -X POST -H "Content-Type: application/json" -d '{"username": ",.input_username>admin", "email": "<input_email>, "password": "<input_password>"}' http://localhost:9202/api/user/token/
```

 - `POST` the following payload to `http://localhost:9202/api/agency/925f12a7-1c44-4c1a-845c-ec8f8975bfc7/batch/route`. Ensure it returns a 201 and inspect the payload in the Django admin.
```json
{"type":"Feature","geometry":{ "type": "MultiLineString",
    "coordinates": [
        [ [100.0, 0.0], [101.0, 1.0] ],
        [ [102.0, 2.0], [103.0, 3.0] ]
  ] },"properties":{"route_id":"43516","route_short_name":"WWW","route_long_name":"W","route_desc":"","route_type":"3","route_url":"http://realtime.catabus.com/InfoPoint/46","route_color":"999900","route_text_color":"FFFFFF","route_sort_order":"21","route_distance":41527.6023805,"trips_monday":0,"trips_tuesday":0,"trips_wednesday":0,"trips_thursday":0,"trips_friday":0,"trips_saturday":0,"trips_sunday":0,"zev_charging_infrastructure":false,"zev_notes":null,"pct_zev_service":null,"num_zev":null,"agency":"925f12a7-1c44-4c1a-845c-ec8f8975bfc7"}}
```

 - `PUT` the following payload to `http://localhost:9202/api/route/batch/62ce2fd7-75b0-467e-9611-aba4dcbb1fa9/` and ensure it returns a 200 and updates the record in the Django admin:
 ```json
{"type":"Feature","geometry":{ "type": "MultiLineString",
    "coordinates": [
        [ [50.0, 0.0], [51.0, 1.0] ],
        [ [52.0, 2.0], [53.0, 3.0] ]
  ] },"properties":{"route_id":"43516","route_short_name":"WWW","route_long_name":"W","route_desc":"","route_type":"3","route_url":"http://realtime.catabus.com/InfoPoint/46","route_color":"999900","route_text_color":"FFFFFF","route_sort_order":"21","route_distance":41527.6023805,"trips_monday":0,"trips_tuesday":0,"trips_wednesday":0,"trips_thursday":0,"trips_friday":0,"trips_saturday":0,"trips_sunday":0,"zev_charging_infrastructure":false,"zev_notes":null,"pct_zev_service":null,"num_zev":null,"agency":"925f12a7-1c44-4c1a-845c-ec8f8975bfc7"}}
```
 - Run `./scripts/tests`
